corehq/apps/es/case_search.py

```diff
@@ -147,6 +147,17 @@ def sort_by_case_property(self, case_property_name, desc=False):
             reset_sort=False
         )
 
+    def sort_by_case_properties(self, sort_properties):
```

Comment:
All of that makes sense to me. Given that this sort is very similar but _slightly different_ from a few other config/criteria options I had some difficulty figuring out the best structure.

> modify `sort_by_case_property` to accept datatype and reset_sort args and use that in `case_search/utils.py` (and use the existing behavior if datatype isn't provided).

I started doing something like this but opted for the other version because it seemed slightly weird to have essentially two different versions of the function within itself (depending on whether `sort_type` and `reset_sort` are passed in). But I think with you're right that it can move to `case_search/utils.py `.

https://github.com/dimagi/commcare-hq/pull/33621/commits/e1d02a9b22f2fc82174a1b846ae921f2b7a4e698

Reply:
> it seemed slightly weird to have essentially two different versions of the function within itself

That makes sense for sure.